File: HTTP_Server_CLI/scenarios/1.2.1-dynamic_3.lua
-- @label: 3 dynamic routes
-- @group: Dynamic routes
-- @competitors: all
-- Round-robins through /user/:id, /post/:slug, /api/v1/:resource.

local counter = 0

request = function()
   counter = counter + 1
   local mod = counter % 3

   if mod == 0 then
      return wrk.format("GET", "/user/" .. (counter % 100))
   elseif mod == 1 then
      return wrk.format("GET", "/post/article-" .. (counter % 100))
   else
      return wrk.format("GET", "/api/v1/resource-" .. (counter % 100))
   end
end

File: HTTP_Server_CLI/scenarios/php/1.2.1-dynamic_3.php
<?php
// @label: 1 dynamic route
// @group: Dynamic routes
// @competitors: all
// Hits /user/:id with rotating IDs.

$paths = [];
for ($i = 0; $i < 100; $i++) {
   $paths[] = "/user/{$i}";
}

return [
   'method' => 'GET',
   'paths'  => $paths,
];
